codebase2/core/U/Core/ABT.hs

```diff
@@ -86,6 +86,24 @@ vmap f (Term _ a out) = case out of
   Cycle r -> cycle a (vmap f r)
   Abs v body -> abs a (f v) (vmap f body)
 
+cata ::
+  Functor f =>
+  (a -> ABT f v x -> x) ->
+  Term f v a ->
+  x
+cata abtAlg =
+  let go (Term _fvs a out) = abtAlg a (fmap go out)
```

Comment:
I'd be cool with chatting about rewriting the base `Term` type as a type alias for Cofree at some point, I think I'll likely want a projection into Cofree when writing structural diffing anyways (although that's a long ways away)